crud_int_test.go

```diff
@@ -0,0 +1,95 @@
+package gocbcorex_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/couchbase/gocbcorex"
+	"github.com/couchbase/gocbcorex/testutilsint"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const NUM_REPLICAS = 2
+
+func TestGetAllReplicasDino(t *testing.T) {
+	opts := CreateDefaultAgentOptions()
+	agent, err := gocbcorex.CreateAgent(context.Background(), opts)
+	defer func() {
+		err := agent.Close()
+		require.NoError(t, err)
+	}()
+
+	require.NoError(t, err)
+
+	docKey := uuid.NewString()
+	upsertRes, err := agent.Upsert(context.Background(), &gocbcorex.UpsertOptions{
+		Key:            []byte(docKey),
+		ScopeName:      "",
+		CollectionName: "",
+		Value:          []byte(`{"foo": "bar"}`),
+	})
+	require.NoError(t, err)
+	assert.NotZero(t, upsertRes.Cas)
+
+	time.Sleep(time.Second * 2)
+
+	ctx := context.Background()
+	stream, err := agent.GetAllReplicas(ctx, &gocbcorex.GetAllReplicasOptions{
+		Key:            []byte(docKey),
+		BucketName:     "default",
+		ScopeName:      "",
+		CollectionName: "",
+	})
+	require.NoError(t, err)
+
+	result, err := stream.Next()
+	masterCount := 0
+	replicaCount := 0
+	for result != nil {
+		require.NoError(t, err)
+		require.Equal(t, result.Value, []byte(`{"foo": "bar"}`))
+		if !result.IsReplica {
+			masterCount += 1
+		} else {
+			replicaCount += 1
+		}
+		result, err = stream.Next()
+	}
+	require.Equal(t, NUM_REPLICAS, masterCount+replicaCount)
+
+	testutilsint.SkipIfNoDinoCluster(t)
```

Comment:
This should be at the top of the test.  If we are doing two "types" of tests (one dino, one not), we should have two tests, one that uses dino and one that doesnt.

Reply:
Done!